controller/array_action/array_mediator_xiv.py

```diff
@@ -163,6 +183,41 @@ def delete_volume(self, volume_id):
 
         logger.info("Finished volume deletion. id : {0}".format(volume_id))
 
+    def get_snapshot(self, snapshot_name):
+        logger.debug("Get snapshot : {}".format(snapshot_name))
+        try:
+            cli_snapshot = self.client.cmd.vol_list(vol=snapshot_name).as_single_element
+        except xcli_errors.IllegalNameForObjectError as ex:
+            logger.exception(ex)
+            raise controller_errors.IllegalObjectName(ex.status)
+        if not cli_snapshot:
+            return None
+        if not cli_snapshot.master_name:
+            raise controller_errors.SnapshotNotFoundVolumeWithSameNameExists(cli_snapshot.name, self.endpoint)
```

Comment:
this error seems very specific (and lacks the word "error").
can we at least shorten it to `VolumeWithSameNameExistsError`?

Reply:
VolumeWithSameName not clear enough. It could be understood that we search for volume and another volume with same name exists.
Will add word "Error"

Reply:
1. is it so bad if we use `VolumeWithSameNameExistsError` for both snapshot and volume searches?
2. is this even relevant for other storage types?
3. what about something like `FoundVolumeInsteadOfSnapshotError`?

Reply:
1. Yes. There are different errors.
2. No. only volume and snapshot is the same object (at least in a9k storage)
3. I prefer my way. It is logner but more informative and clear.

Reply:
1. then what about `VolumeWithSameNameExistsSnapshotError`/`SnapshotNameFoundIsVolumeError`?
2. if this is only relevant to a9k, we might want a more general error here (since these errors are not storage-specific)
3. this shortening is better than nothing, but I'm sure we can find something better!

Reply:
Ok SnapshotNameBelongsToVolumeError